crates/harp/src/parse.rs

```diff
@@ -76,9 +79,24 @@ pub fn parse_exprs_ext<'a>(input: &ParseInput<'a>) -> crate::Result<RObject> {
             code: parse_input_as_string(input).unwrap_or(String::from("Conversion error")),
             message: String::from("Incomplete code"),
         }),
+        ParseResult::SyntaxError { message, line } => {
+            Err(crate::Error::ParseSyntaxError { message, line })
+        },
     }
 }
 
+// Return type would be clearer if syntax error was integrated in `ParseResult`
+pub fn parse_with_parse_data(text: &str) -> crate::Result<(ParseResult, ParseData)> {
+    let srcfile = srcref::SrcFile::try_from(text)?;
+
+    // Fill parse data in `srcfile` by side effect
+    let status = parse_status(&ParseInput::SrcFile(&srcfile))?;
+
+    let parse_data = ParseData::from_srcfile(&srcfile)?;
+
+    Ok((status, parse_data))
+}
```

Comment:
- This method does not seem to be used
- The leading comment about `Return type would be clearer` seems outdated because syntax error is integrated into `ParseResult` now I think

Reply:
It's no longer used, neither is `ParseData`, but I've left those for later in case they are needed. E.g. maybe it'll be useful to generate rowan trees from the R parser in a more efficient way than traversing R AST.